core/unmarshal_v2_primitives_test.go

```diff
@@ -0,0 +1,1386 @@
+
+	err = UnmarshalPrimitive(rawMap, "prop_map_slice", &model.PropMapSlice)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropMapSlice)
+	assert.Equal(t, d1, model.PropMapSlice[0]["key1"].String())
+	assert.Equal(t, d2, model.PropMapSlice[1]["key2"].String())
+
+	err = UnmarshalPrimitive(rawMap, "prop_slice_map_slice", &model.PropSliceMapSlice)
+	assert.Nil(t, err)
+	assert.NotNil(t, model.PropSliceMapSlice)
+	assert.Equal(t, d1, model.PropSliceMapSlice[0]["key1"][0].String())
+	assert.Equal(t, d2, model.PropSliceMapSlice[1]["key2"][0].String())
+	assert.Equal(t, d3, model.PropSliceMapSlice[1]["key2"][1].String())
+	assert.Equal(t, d4, model.PropSliceMapSlice[1]["key2"][2].String())
+
+	// Tests involving a JSON null value
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.Prop)
+	assert.Nil(t, err)
+	assert.Nil(t, model.Prop)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSlice)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropMap)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropMap)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSliceMap)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSliceMap)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropMapSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropMapSlice)
+
+	err = UnmarshalPrimitive(rawMap, "null_prop", &model.PropSliceMapSlice)
+	assert.Nil(t, err)
+	assert.Nil(t, model.PropSliceMapSlice)
+
+	// Negative tests
+	err = UnmarshalPrimitive(rawMap, "bad_type", &model.Prop)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(err.Error(), "error unmarshalling property 'bad_type'"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	err = UnmarshalPrimitive(rawMap, "not_a_slice", &model.PropSlice)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(err.Error(), "error unmarshalling property 'not_a_slice'"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	err = UnmarshalPrimitive(rawMap, "bad_slice_type", &model.PropSlice)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(err.Error(), "error unmarshalling property 'bad_slice_type'"))
+	t.Logf("Expected error: %s\n", err.Error())
+}
+
+func TestUnmarshalPrimitiveDateTime(t *testing.T) {
+	type MyModel struct {
+		Prop              *strfmt.DateTime
+		PropSlice         []strfmt.DateTime
+		PropMap           map[string]strfmt.DateTime
+		PropSliceMap      map[string][]strfmt.DateTime
+		PropMapSlice      []map[string]strfmt.DateTime
+		PropSliceMapSlice []map[string][]strfmt.DateTime
+	}
+
+	jsonTemplate := `{
+		"prop": "%d1",
+		"prop_slice": ["%d1", "%d2"],
+		"prop_map": { "key1": "%d1", "key2": "%d2" },
+		"prop_slice_map": { "key1": ["%d1", "%d2"], "key2": ["%d3", "%d4"] },
+		"prop_map_slice": [{"key1": "%d1"}, {"key2": "%d2"}],
+		"prop_slice_map_slice": [{"key1": ["%d1"]}, {"key2": ["%d2", "%d3", "%d4"]} ],
+		
+		"bad_type":  true,
+		"not_a_slice": false,
+		"bad_slice_type": [38, 26],
+		"null_prop": null
```

Comment:
it turns out that "" will be unmarshalled by the strfmt.DateTime unmarshalling code to produce "1970-01-01T00:00:00.000Z", rather than produce an error.

Reply:
Ouch!